Lib/ufo2ft/featureCompiler.py

```diff
@@ -58,19 +58,22 @@ def setupFile_features(self):
 
         existing = self.font.features.text or ""
 
-        # build the GPOS features as necessary
+        # build features as necessary
         autoFeatures = {}
-        if "kern" not in features:
-            autoFeatures["kern"] = self.writeFeatures_kern()
         # the current MarkFeatureWriter writes both mark and mkmk features
         # with shared markClass definitions; to prevent duplicate glyphs in
         # markClass, here we write the features only if none of them is alread
         # present.
         # TODO: Support updating pre-existing markClass definitions to allow
         # writing either mark or mkmk features indipendently from each other
         # https://github.com/googlei18n/fontmake/issues/319
-        if "mark" not in features and "mkmk" not in features:
-            autoFeatures["mark"] = self.writeFeatures_mark()
+        for featureWriter in self.featureWriters:
+            doFeatures = []
+            for feature in featureWriter.features:
+                if feature not in features:
+                    doFeatures.append(feature)
+            if set(doFeatures) == set(featureWriter.features):
```

Comment:
maybe you could replace this `doFeatures` for loop and subsequent `if` test with
```python
if not any(fea for fea in featureWriter.features if fea in features):
```

Reply:
Yeah that’s better.
But I want to add a flag `skip_if_features_are_present` (or a better name ;-) to the featureWriter so the featureCompiler can still add the feature block even if one is already present. For example, if I have a kern block with contextual kerning lookups, I still want the KernFeatureWrite to add the pair kerning lookups.

Reply:
> I want to add a flag skip_if_features_are_present

maybe you could use an enum rather than a boolean option to control the behavior of the feature writers. One may want to do one of these three : skip if present (current default), overwrite, or append.